src/test/java/bio/terra/service/dataset/DatasetRequestValidatorTest.java

```diff
@@ -168,6 +168,32 @@ public void testDuplicateTableNames() throws Exception {
         });
   }
 
+  /**
+   * Modify and return the request so that it has a single table with the specified name, no
+   * relationships, and no assets.
+   */
+  private DatasetRequestModel withNamedTable(DatasetRequestModel request, String tableName) {
+    ColumnModel column = new ColumnModel().name("id").datatype(TableDataType.STRING);
+    TableModel table = new TableModel().name(tableName).columns(List.of(column));
+    request.getSchema().tables(List.of(table)).relationships(List.of()).assets(List.of());
+
+    return request;
+  }
+
+  @Test
+  public void testInvalidTableName() throws Exception {
+    DatasetRequestModel req = buildDatasetRequest();
+
+    // Table names with leading underscores are invalid
+    checkValidationErrorModel(
+        expectBadDatasetCreateRequest(withNamedTable(req, "_")), new String[] {"Pattern"});
+
+    // Table names over 63 characters are invalid
+    checkValidationErrorModel(
+        expectBadDatasetCreateRequest(withNamedTable(req, StringUtils.repeat("a", 64))),
```

Comment:
There is no need to use `StringUtils` here since `String` has a `repeat` method:
```suggestion
        expectBadDatasetCreateRequest(withNamedTable(req, "a".repeat(64))),
```